tests/Rule/NonStaticAssertionsTest.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of DOCtor-RST.
+ *
+ * (c) Oskar Stark <oskarstark@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Tests\Rule;
+
+use App\Rule\NonStaticAssertions;
+use App\Tests\RstSample;
+use App\Tests\UnitTestCase;
+use App\Value\NullViolation;
+use App\Value\Violation;
+use App\Value\ViolationInterface;
+
+final class NonStaticAssertionsTest extends UnitTestCase
+{
+    /**
+     * @test
+     *
+     * @dataProvider checkProvider
+     */
+    public function check(ViolationInterface $expected, RstSample $sample): void
+    {
+        self::assertEquals(
+            $expected,
+            (new NonStaticAssertions())->check($sample->lines, $sample->lineNumber, 'filename'),
+        );
+    }
+
+    public static function checkProvider(): iterable
+    {
+        foreach (self::phpCodeBlocks() as $codeBlock) {
+        
+            yield [
+                Violation::from(
+                    'Please use `$this->assert` over static call',
+                    'filename',
+                    1,
+                    'static::assertFalse($expirationChecker->isExpired($validUntil));',
+                ),
+                new RstSample([
+                    $codeBlock,
+                    'static::assertFalse($expirationChecker->isExpired($validUntil));'
```

Comment:
```suggestion
                    '    static::assertFalse($expirationChecker->isExpired($validUntil));'
```

Same for the others, otherwise it is on the same level like the code block and not part of the code block